kobo/apps/stripe/tests/test_organization_usage.py

```diff
@@ -0,0 +1,186 @@
+            current_period_start=self.now - relativedelta(weeks=2),
+        )
+
+    def test_usage_doesnt_include_org_users_without_subscription(self):
+        """
+        Test that the endpoint *only* returns usage for the logged-in user
+        if they don't have a subscription that includes Organizations.
+        """
+        response = self.client.get(self.detail_url)
+        # without a plan, the user should only see their usage
+        assert response.data['total_submission_count']['all_time'] == self.expected_submissions_single
+        assert response.data['total_submission_count']['current_month'] == self.expected_submissions_single
+        assert response.data['total_storage_bytes'] == (
+            self.expected_file_size() * self.expected_submissions_single
+        )
+
+    def test_usage_for_plans_with_org_access(self):
+        """
+        Test that the endpoint aggregates usage for each user in the organization
+        when viewing /service_usage/{organization_id}/
+        """
+
+        self.generate_subscription(
+            {
+                'plan_type': 'enterprise',
+                'organizations': True,
+            }
+        )
+
+        # the user should see usage for everyone in their org
+        response = self.client.get(self.detail_url)
+        assert response.data['total_submission_count']['current_month'] == self.expected_submissions_multi
+        assert response.data['total_submission_count']['all_time'] == self.expected_submissions_multi
+        assert response.data['total_storage_bytes'] == (
+            self.expected_file_size() * self.expected_submissions_multi
+        )
+
+    def test_doesnt_include_org_users_with_invalid_plan(self):
+        """
+        Test that the endpoint *doesn't* aggregates usage for the organization
+        when subscribed to a product that doesn't include org access
+        """
+
+        self.generate_subscription({})
+
+        response = self.client.get(self.detail_url)
+        # without the proper subscription, the user should only see their usage
+        assert response.data['total_submission_count']['current_month'] == self.expected_submissions_single
+        assert response.data['total_submission_count']['all_time'] == self.expected_submissions_single
+        assert response.data['total_storage_bytes'] == (
+            self.expected_file_size() * self.expected_submissions_single
+        )
+
+    def test_endpoint_speed(self):
+        # get the average request time for 10 hits to the endpoint
+        single_user_time = timeit.timeit(lambda: self.client.get(self.detail_url), number=10)
+
+        self.generate_subscription(
+            {
+                'plan_type': 'enterprise',
+                'organizations': True,
+            }
+        )
+
+        # get the average request time for 10 hits to the endpoint
+        multi_user_time = timeit.timeit(lambda: self.client.get(self.detail_url), number=10)
+        assert single_user_time < 1.5
```

Comment:
This timing logic dangerous for CI. Please make it skip by default, fine to leave the test in for manual running only. CI can often be slower and it makes the test non-deterministic.